### DISCUSSION TITLE
Certificate export formats

### DISCUSSION THREAD

**Discussion by jortaca** [PeculiarVentures/fortify#539] [General]
Hello, Is it possible to export a selected certificate in p12 or crt format?

Thanks!

**Comment by microshine** [comment]
Yes, it's possible to export cert from the token. Please use this example https://github.com/PeculiarVentures/webcrypto.dev-examples/blob/e43a32afc102f3a6a67721ae3ee0e3978763a009/src/crypto/fortify/cert_list.ts#L65

You just need to get a cert raw and save it in required format. You can use @peculiar/x509 library to convert a certificate raw into the PEM format

**Reply by jortaca** [reply]
Ok, but this doesn't contain the private key, does it?

**Reply by microshine** [reply]
You are right. It doesn't have a private key. I think this is not a good idea to export the private key from the token. If your key is extractable, you can export it using WebCrypto API

```
const pkcs8 = await crypto.subtle.exportKey("pkcs8", privateKey);
```

If you need to make a PFX file (with certificate and its private key) you can use extracted raw data and lib like `pkijs`

https://github.com/PeculiarVentures/PKI.js/blob/469c403d102ee5149e8eb9ad19754c9696ed7c55/test/nodePKCS12Example.spec.ts#L15

**Reply by jortaca** [reply]
Thank you, I'll try to do it that way.
